projects/igniteui-angular/src/lib/tabs/tabs.directive.ts

```diff
@@ -196,33 +200,32 @@ export abstract class IgxTabsDirective extends IgxCarouselComponentBase implemen
     protected onItemChanges() {
         this.setAttributes();
 
-        if (this.selectedIndex >= 0 && this.selectedIndex < this.items.length) {
+        // Check if there is selected tab
+        let selectedIndex = -1;
+        this.items.some((tab, i) => {
+            if (tab.selected) {
+                selectedIndex = i;
+            }
+            return tab.selected;
+        });
 
-            // Check if there is selected tab
-            let selectedIndex = -1;
-            this.items.some((tab, i) => {
-                if (tab.selected) {
-                    selectedIndex = i;
-                }
-                return tab.selected;
+        if (selectedIndex >= 0) {
+            // Set the selected index to the tab that has selected=true
+            Promise.resolve().then(() => {
+                this.selectedIndex = selectedIndex;
             });
-
-            if (selectedIndex >= 0) {
-                // Select the same tab that was previously selected
-                Promise.resolve().then(() => {
-                    this.selectedIndex = selectedIndex;
-                });
-            } else {
+        } else {
+            if (this.selectedIndex >= 0 && this.selectedIndex < this.items.length) {
                 // Select the tab on the same index the previous selected tab was
                 Promise.resolve().then(() => {
                     this.updateSelectedTabs(null);
                 });
+            } else if (this.selectedIndex >= this.items.length) {
+                // Select the last tab
+                Promise.resolve().then(() => {
+                    this.selectedIndex = this.items.length - 1;
+                });
             }

```

Comment:
I believe we should also check for -1 and if that's the case we should set selectedIndex to 0;
```suggestion
          } else if (this.selectedIndex >= this.items.length) {
                // Select the last tab
                Promise.resolve().then(() => {
                    this.selectedIndex = this.items.length - 1;
                });
            } else {
                Promise.resolve().then(() => {
                    this.selectedIndex = 0;
                });
            }

```